js/main.js

```diff
@@ -195,3 +200,43 @@ function onConfirmDeleteClick(event) {
 }
 
 $confirmBtn.addEventListener('click', onConfirmDeleteClick);
+
+var $searchBtn = document.querySelector('.searchBtn');
+var $searchBar = document.querySelector('.searchBar');
+var $noMatches = document.querySelector('.noMatches');
+
+function onSearch(event) {
+  switchViews('entries');
+  var searchValue = $searchBar.value.toLowerCase();
+  var $domEntriesList = document.querySelectorAll('.entryList li');
+  var toView = [];
+  $noMatches.className = 'hidden';
+
+  for (var domEntry = 0; domEntry < $domEntriesList.length; domEntry++) {
+    var $currDomEntry = $domEntriesList[domEntry];
+    var $currTitle = $currDomEntry.querySelector('h2').textContent;
+    var $currNotes = $currDomEntry.querySelector('p').textContent;
+    var $currEntryText = ($currTitle + ' ' + $currNotes).toLowerCase();
+    var currWord = '';
+    $currDomEntry.className = '';
+
+    for (var letterInText = 0; letterInText < $currEntryText.length; letterInText++) {
+      if ($currEntryText[letterInText] === ' ' || $currEntryText[letterInText] === '.' || $currEntryText[letterInText] === ',') {
+        if (currWord === searchValue) {
+          toView.push($currDomEntry);
+        }
+        currWord = '';
+      } else {
+        currWord += $currEntryText[letterInText];
+      }
+    }
+    if (!toView.includes($currDomEntry)) {
```

Comment:
I think you can forego this loop entirely if you change `!toView.includes($currDomEntry)` to `!$currEntryText.includes(searchValue)` in your conditional here